balancer/endpointsharding/endpointsharding.go

```diff
@@ -68,26 +48,26 @@ type ChildState struct {
 	Balancer balancer.ExitIdler
 }
 
-// NewBalancer returns a load balancing policy that manages homogeneous child
-// policies each owning a single endpoint. The balancer will automatically call
-// ExitIdle on its children if they report IDLE connectivity state.
-func NewBalancer(cc balancer.ClientConn, opts balancer.BuildOptions) balancer.Balancer {
-	return newBlanacer(cc, opts, true)
-}
-
-// NewBalancerWithoutAutoReconnect returns a load balancing policy that manages
-// homogeneous child policies each owning a single endpoint. The balancer will
-// allow children to remain in IDLE state until triggered to exit idle state
-// using the ChildState obtained using the endpointsharding picker.
-func NewBalancerWithoutAutoReconnect(cc balancer.ClientConn, opts balancer.BuildOptions) balancer.Balancer {
-	return newBlanacer(cc, opts, false)
+// Options are the options to configure the behaviour of the
+// endpointsharding balancer.
+type Options struct {
+	// DisableAutoReconnect allows the balancer to keep child balancer in the
+	// IDLE state until they are explicitly triggered to exit using the
+	// ChildState obtained from the endpointsharding picker. When set to false,
+	// the endpointsharding balancer will automatically call ExitIdle on child
+	// connections that report IDLE.
+	DisableAutoReconnect bool
 }
 
-func newBlanacer(cc balancer.ClientConn, opts balancer.BuildOptions, autoReconnect bool) balancer.Balancer {
+// NewBalancer returns a load balancing policy that manages homogeneous child
+// policies each owning a single endpoint. The endpointsharding balancer
+// forwards the LoadBalancingConfig in ClientConn state updates to its children.
+func NewBalancer(cc balancer.ClientConn, opts balancer.BuildOptions, childBuilder balancer.Builder, esOpts Options) balancer.Balancer {
```

Comment:
I'm wondering about the signature here.

There's a lot of options:

1. Keep this & make anyone wanting to use es+lazy wrap lazy in a Builder implementation.
2. Keep this & implement a Builder in `lazy`, still.
3. Create `type ChildBuilder interface { Build(..) .. }` without `Name()` and make `lazy` implement that.
4. Accept a function matching the signature of `Build` here.

None of these are great.  I think I might like (4) the best for usability?  But I dislike it for understandability.  I suppose that could be helped with:

```go
type ChildBuilder = func(...) ... // matching Build
```

?

Reply:
Updated to use a type declaration instead of an alias.